packages/strapi/lib/commands/generate-template.js

```diff
@@ -0,0 +1,94 @@
+'use strict';
+
+const { resolve, join, basename } = require('path');
+const fse = require('fs-extra');
+const chalk = require('chalk');
+const inquirer = require('inquirer');
+
+// All directories that a template could need
+const TEMPLATE_CONTENT = ['api', 'components', 'config/functions/bootstrap.js', 'data'];
+
+/**
+ *
+ * @param {string} templatePath Absolute path to template directory
+ * @param {string} rootBase Name of the root directory
+ */
+async function copyContent(templatePath, rootBase) {
+  for (const item of TEMPLATE_CONTENT) {
+    try {
+      await fse.copy(join(process.cwd(), item), join(templatePath, item));
+      const currentProjectBase = basename(process.cwd());
+      console.log(
+        `${chalk.green(
+          'success'
+        )}: copy ${currentProjectBase}/${item} => ${rootBase}/template/${item}`
+      );
+    } catch (error) {
+      console.error(`${chalk.red('error')}: ${error.message}`);
+    }
+  }
+}
+
+/**
+ *
+ * @param {string} rootPath Absolute path to the root directory
+ */
+async function writeTemplateJson(rootPath) {
+  try {
+    await fse.writeJSON(join(rootPath, 'template.json'), {});
+    console.log(`${chalk.green('success')}: create JSON config file`);
+  } catch (error) {
+    console.error(`${chalk.red('error')}: ${error.message}`);
+  }
+}
+
+/**
+ *
+ * @param {string} rootPath Absolute path to the root directory
+ * @returns boolean
+ */
+async function templateConfigExists(rootPath) {
+  const jsonConfig = await fse.pathExists(join(rootPath, 'template.json'));
+  const functionConfig = await fse.pathExists(join(rootPath, 'template.js'));
+
+  return jsonConfig || functionConfig;
+}
+
+module.exports = async function generateTemplate(directory) {
+  // Allow any relative path,
+  // otherwise default destination is at the same level as the current directory
+  const dir = directory.startsWith('.') ? directory : `../${directory}`;
```

Comment:
I'm not sure this is necessary to create a folder outisde of the project by default 🤔  if the cli is well documented then the user will know the directory is the path the template will be saved to so they should decide on their own don't you think ?

Reply:
Yeah I see what you mean.  I think since I originally defined the experience as providing the name and defaulting to a path, I was sort of stuck on that idea.  But it definitely makes more sense to just let the user have full control.  I'll update.